master/buildbot/steps/source/p4.py

```diff
@@ -231,12 +232,18 @@ def _buildVCCommand(self, doCommand):
         command.extend(doCommand)
 
         def encodeArg(arg):
-            if isinstance(arg, string_types):
+            if isinstance(arg, binary_type):
```

Comment:
I don't see how this function can be moved into a general utility.
The logic in here is pretty specific to this module, such as for a tuple,
encoding only the second tuple argument